src/main/java/org/apache/commons/io/input/BOMInputStream.java

```diff
@@ -198,10 +207,16 @@ public boolean hasBOM() throws IOException {
      * @throws IOException
      *             if an error reading the first bytes of the stream occurs
      */
+    // getBOM() returns byteOrderMark. In following function null check is performed for 
+    // byteOrderMark and after that getBOM() is invoked.  This warning is false positive
+    // and correctness cannot be established due to typechecking limitations in checker. 
+    @SuppressWarnings("nullness:dereference.of.nullable")
     public boolean hasBOM(final ByteOrderMark bom) throws IOException {
         if (!boms.contains(bom)) {
             throw new IllegalArgumentException("Stream not configure to detect " + bom);
         }
+        // TODO: The first clause uses byteOrderMark, and the second clause uses getBom().
+        // The inconsistency is confusing.  Both clauses should use the same expression.
```

Comment:
@mernst  `getBOM()` can return null in case specified `bom` is not present, I misinterpreted `bom` parameter with `byteordermark`. My comment is false and this might be possible bug. 
```
public boolean hasBOM(final ByteOrderMark bom) throws IOException {
        if (!boms.contains(bom)) {
            throw new IllegalArgumentException("Stream not configure to detect " + bom);
        }
        return byteOrderMark != null && getBOM().equals(bom);
}
```  
The only addition required is adding necessary null check for `getBOM()` before using it in return statement. I did not have any solid test-case, but this is important. Can I make this addition and send a pull request.

Reply:
I think this was actually my mistake rather than yours, but thanks for noticing it.  Can you create a test case that crashes with a null pointer exception?  If so, definitely submit a bug report and proposed patch.